caluma/extensions/tests/test_visibilities.py

```diff
@@ -1,16 +1,91 @@
-from caluma.caluma_form import models, schema
+import pytest
+
+from caluma.caluma_form import models as form_models, schema as form_schema
+from caluma.caluma_workflow import models as workflow_models, schema as workflow_schema
 from caluma.extensions.visibilities import MySAGWVisibility
 
 
+@pytest.mark.parametrize(
+    "name,visible_map",
+    [
+        (
+            "admin",
+            {
+                "Form": 8,
+                "WorkItem": 2,
+                "Document": 6,
+                "Case": 4,
+                "Question": 1,
+                "Answer": 1,
+            },
+        ),
+        (
+            "test1",
+            {
+                "Form": 8,
+                "WorkItem": 0,
+                "Document": 1,
+                "Case": 1,
+                "Question": 1,
+                "Answer": 1,
+            },
+        ),
+        (
+            "test2",
+            {
+                "Form": 8,
+                "WorkItem": 1,
+                "Document": 0,
+                "Case": 1,
+                "Question": 1,
+                "Answer": 0,
+            },
+        ),
+    ],
+)
 def test_visibilities_default(
     db,
-    form,
+    work_item_factory,
+    form_question_factory,
+    answer_factory,
     admin_info,
+    name,
+    visible_map,
 ):
-    vis = MySAGWVisibility()
+    admin_info.context.user.username = name
+    admin_info.context.user.groups = [name]
+
+    wi1 = work_item_factory(
+        assigned_users=["foo"],
+        document__created_by_user="test1",
+        case__created_by_user="test2",
+    )
+    work_item_factory(
+        assigned_users=["test2"],
+        document__created_by_user="foo",
+        case__created_by_user="test1",
+    )
 
-    queryset = models.Form.objects.all()
+    fq = form_question_factory(form=wi1.document.form, question__type="text")
+    answer_factory(
+        document=wi1.document,
+        question=fq.question,
+        value="foo",
+        created_by_user="test1",
+    )
 
-    filtered = vis.filter_queryset(schema.Form, queryset, admin_info)
+    test_map = [
```

Comment:
This could be parametrized instead, no? Might be a little bit slower though, but probably more readable.

Reply:
As discussed in person.